server/src/main/java/org/jetbrains/maven/project/MavenProjectWrapper.java

```diff
@@ -23,12 +25,42 @@ public static MavenProjectWrapper fromBase(URI projectBase) {
     return new MavenProjectWrapper(result, projectBase);
   }
 
-  public List<BuildTargetIdentifier> getDependencies() {
+  public List<BuildTargetIdentifier> getDependencies(Map<ProjectId, MavenProjectWrapper> modules) {
     MavenProject project = getProject();
-    return project.getDependencies().stream()
-        .map(Dependency::getManagementKey)
-        .map(BuildTargetIdentifier::new)
-        .collect(Collectors.toList());
+    List<BuildTargetIdentifier> dependencies = new ArrayList<>();
+    for (Dependency dependency : project.getDependencies()) {
+      ProjectId projectId = new ProjectId(
+          dependency.getGroupId(),
+          dependency.getArtifactId(),
+          dependency.getVersion()
+      );
+      if (modules.containsKey(projectId)) {
```

Comment:
it's good, but like I said, it's not obligatory - client should handle all (including external) dependencies as well